src/components/ChatApp/MessageComposer.react.js

```diff
@@ -151,103 +220,121 @@ class MessageComposer extends Component {
       rows: 1,
       recognizing: false,
     });
-    flag = 0;
+    this.flag = 0;
   };
 
   onResult = ({ interimTranscript, finalTranscript }) => {
     if (interimTranscript === undefined) {
-      let result = finalTranscript;
       this.setState({
-        result: result,
+        result: finalTranscript,
         color: '#ccc',
       });
     } else {
-      let result = interimTranscript;
       this.setState({
-        result: result,
+        result: interimTranscript,
         color: '#ccc',
       });
       if (finalTranscript) {
-        result = finalTranscript;
         this.setState({
-          result: result,
+          result: finalTranscript,
           color: '#000',
         });
         this.speakDialogClose();
       }
     }
   };
 
-  componentDidMount() {
-    let micInputSetting = UserPreferencesStore.getMicInput();
-    if (micInputSetting) {
-      // Getting the Speech Recognition to test whether possible
-      const SpeechRecognition =
-        window.SpeechRecognition ||
-        window.webkitSpeechRecognition ||
-        window.mozSpeechRecognition ||
-        window.msSpeechRecognition ||
-        window.oSpeechRecognition;
-      // Setting buttons accordingly
-      if (SpeechRecognition != null) {
-        this.Button = <Mic />;
-        this.speechRecog = true;
+  _onClickButton = () => {
+    const { text, recognizing } = this.state;
+    const { threadID, speechOutputAlways } = this.props;
+    this.flag = 1;
+    if (text === '') {
+      if (this.speechRecog) {
+        this.setState({ start: true });
       } else {
-        this.Button = <Send />;
-        this.speechRecog = false;
+        this.setState({ start: false });
       }
     } else {
-      this.Button = <Send />;
-      this.speechRecog = false;
-    }
-
-    let testSkill = $.urlParam('testExample');
-    if (testSkill) {
-      let text = testSkill.trim();
+      let text = text.trim();
       if (text) {
-        let enterAsSend = UserPreferencesStore.getEnterAsSend();
-        if (!enterAsSend) {
+        let EnterAsSend = UserPreferencesStore.getEnterAsSend();
```

Comment:
I think you should let the first letter small.
`EnterAsSend` to `enterAsSend`

Reply:
@samagragupta Those functions have been defined in a different component, so I thought I should not mess with their names. I know camel case should be used but seems like the previous devs forgot to use it.

Reply:
Make it camelCase. The correction needs to made in other files while refactoring them.